cli/golem-cli/test-data/rust-code-first-snippets/lib.rs

```diff
@@ -0,0 +1,405 @@
+//         self.bar_agent.fun_simple_enum(simple_enum)
+//     }
+//
+//     fn fun_complex_enum(&self, complex_enum: ComplexEnumType) -> ComplexEnumType {
+//         self.bar_agent.fun_complex_enum(complex_enum)
+//     }
+//
+//     fn fun_simple_struct(&self, simple_struct: SimpleStructType) -> SimpleStructType {
+//         self.bar_agent.fun_simple_struct(simple_struct)
+//     }
+//
+//     fn fun_result(&self, result: Result<String, String>) -> Result<String, String> {
+//         self.bar_agent.fun_result(result)
+//     }
+// }
+
+#[agent_definition]
+trait BarAgent {
+    fn new(opt_string: Option<String>) -> Self;
+
+    // fn fun_all(
+    //     &self,
+    //     struct_complex: StructComplexType,
+    //     union_complex: EnumComplexType,
+    //     f64: f64,
+    //     f32: f32,
+    //     u32: u32,
+    //     i32: i32,
+    //     ui6: u16,
+    //     i16: i16,
+    //     i8: i8,
+    //     u8: u8,
+    //     string: String,
+    //     bool: bool,
+    //     map: MapType,
+    //     tuple_complex: TupleComplexType,
+    //     tuple: TupleType,
+    //     list_complex: ListOfObjectType,
+    //     list_string: ListOfStringType,
+    //     enum_with_only_literals: EnumWithOnlyLiterals,
+    //     simple_enum: SimpleEnumType,
+    //     complex_enum: ComplexEnumType,
+    //     simple_struct: SimpleStructType,
+    //     struct_with_single_field: StructWithSingleField,
+    //     optional_string: OptionalStringType,
+    // ) -> String;
+
+    // fn fun_optional(
+    //     &self,
+    //     param1: Option<String>,
+    //     param2: Option<StructComplexType>,
+    // ) -> Option<String>;
+    //
+    // fn fun_no_return(&self, text: String);
+    //
+    // fn fun_number(&self, number: f64) -> f64;
+
+    fn fun_string(&self, string: String) -> String;
+
+    fn fun_mut(&mut self, string: String) -> String;
+
+    // fn fun_boolean(&self, boolean: bool) -> bool;
+    //
+    // fn fun_map(&self, map: MapType) -> MapType;
+    //
+    // fn fun_struct_complex(&self, complex: StructComplexType) -> StructComplexType;
+    //
+    // fn fun_tuple_complex(&self, complex: TupleComplexType) -> TupleComplexType;
+    //
+    // fn fun_tuple(&self, tuple: TupleType) -> TupleType;
+    //
+    // fn fun_list_complex(&self, list_complex: ListOfObjectType) -> ListOfObjectType;
+    //
+    // fn fun_list_string(&self, list_string: ListOfStringType) -> ListOfStringType;
+    //
+    // fn fun_enum_with_only_literals(
+    //     &self,
+    //     enum_with_only_literals: EnumWithOnlyLiterals,
+    // ) -> EnumWithOnlyLiterals;
+    //
+    // fn fun_simple_enum(&self, simple_enum: SimpleEnumType) -> SimpleEnumType;
+    //
+    // fn fun_complex_enum(&self, complex_enum: ComplexEnumType) -> ComplexEnumType;
+    //
+    // fn fun_simple_struct(&self, simple_struct: SimpleStructType) -> SimpleStructType;
+    //
+    // fn fun_result(&self, result: Result<String, String>) -> Result<String, String>;
```

Comment:
Currently the commented code doesn't compile in agent code.. As more tasks get finished, these can be uncommented. Once RPC is implemented, `FooAgent` will be uncommented, and once type support is done, all of this should be uncommented